### PR TITLE
Update compute api client and AutomaticRestart usage

### PR DESCRIPTION
**Pull request by selmanj**
Instance.AutomaticRestart changed from a bool to a *bool to distinguish between
an empty value vs an explicitly set value. See
https://github.com/google/google-api-go-client/issues/196 for more context.

I'm running the acceptance test `compute*` suite to verify nothing breaks. Motivation for updating the api client is to use the newly-launched `Labels` field inside a compute struct.

### CODE REVIEW

File: google/resource_compute_instance_template.go [hunk 1]
```diff
@@ -679,13 +680,16 @@ func flattenNetworkInterfaces(networkInterfaces []*compute.NetworkInterface) ([]
...
 	result := make([]map[string]interface{}, 0, 1)
 	schedulingMap := make(map[string]interface{})
-	schedulingMap["automatic_restart"] = scheduling.AutomaticRestart
+	if scheduling.AutomaticRestart != nil {
+		schedulingMap["automatic_restart"] = &scheduling.AutomaticRestart
+	}
 	schedulingMap["on_host_maintenance"] = scheduling.OnHostMaintenance
 	schedulingMap["preemptible"] = scheduling.Preemptible
 	result = append(result, schedulingMap)
+	// TODO(jsselman) No need to return two values as automatic restart is captured in map
```
Comment (danawillow) [inline google/resource_compute_instance_template.go]: Technically you're selmanj here :)

Comment (selmanj) [inline google/resource_compute_instance_template.go]: argh

**Comment by danawillow** [comment]
Seems reasonable. Tentative LGTM pending test results

**Comment by rileykarson** [comment]
LGTM if the tests are good.

File: google/resource_compute_instance_template.go [hunk 2]
```diff
@@ -537,7 +538,7 @@ func resourceComputeInstanceTemplateCreate(d *schema.ResourceData, meta interfac
 		_scheduling := _schedulings[0].(map[string]interface{})
 
 		if vp, okp := _scheduling["automatic_restart"]; okp {
-			instanceProperties.Scheduling.AutomaticRestart = vp.(bool)
+			instanceProperties.Scheduling.AutomaticRestart = googleapi.Bool(vp.(bool))
 			forceSendFieldsScheduling = append(forceSendFieldsScheduling, "AutomaticRestart")
```
Comment (rileykarson) [inline google/resource_compute_instance_template.go]: Nit: We shouldn't need this anymore

**Comment by selmanj** [comment]
Tests failed. Looks like I used a & where I meant *

Example failure here. Will correct
```
=== RUN   TestAccComputeBackendService_withBackend
--- FAIL: TestAccComputeBackendService_withBackend (12.10s)
	testing.go:428: Step 0 error: Error applying: 1 error(s) occurred:

		* google_compute_instance_template.foobar: 1 error(s) occurred:

		* google_compute_instance_template.foobar: Error setting scheduling: scheduling.0.automatic_restart: '' expected type 'bool', got unconvertible type '**bool'
	testing.go:492: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Error refreshing: 1 error(s) occurred:

		* google_compute_instance_template.foobar: 1 error(s) occurred:

		* google_compute_instance_template.foobar: google_compute_instance_template.foobar: Error setting scheduling: scheduling.0.automatic_restart: '' expected type 'bool', got unconvertible type '**bool'

		State: <nil>
	resource_compute_backend_service_test.go:66: Expected TimeoutSec == 10, got 0
	resource_compute_backend_service_test.go:69: Expected Protocol to be HTTP, got ""
	resource_compute_backend_service_test.go:72: Expected 1 backend, got 0
```

Also am getting some quota errors. Will request higher quota.
```
=== RUN   TestAccComputeRoute_basic
--- FAIL: TestAccComputeRoute_basic (0.52s)
	testing.go:428: Step 0 error: Error applying: 1 error(s) occurred:

		* google_compute_network.foobar: 1 error(s) occurred:

		* google_compute_network.foobar: Error creating network: googleapi: Error 403: Quota 'NETWORKS' exceeded. Limit: 5.0, quotaExceeded
```

**Comment by selmanj** [comment]
No errors, but the tests timed out after two hours. This is just running tests with the prefix 'TestAccCompute'. Is it normal for it to be this slow?

```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccCompute* -timeout 120m
?   	github.com/terraform-providers/terraform-provider-google	[no test files]
=== RUN   TestAccComputeImage_resolveImage
--- PASS: TestAccComputeImage_resolveImage (83.96s)
=== RUN   TestAccComputeAddress_importBasic
--- PASS: TestAccComputeAddress_importBasic (23.45s)
=== RUN   TestAccComputeAutoscaler_importBasic
--- PASS: TestAccComputeAutoscaler_importBasic (70.50s)
=== RUN   TestAccComputeBackendService_importBasic
--- PASS: TestAccComputeBackendService_importBasic (44.85s)
=== RUN   TestAccComputeDisk_importBasic
--- PASS: TestAccComputeDisk_importBasic (24.55s)
=== RUN   TestAccComputeFirewall_importBasic
--- PASS: TestAccComputeFirewall_importBasic (66.82s)
=== RUN   TestAccComputeForwardingRule_importBasic
--- PASS: TestAccComputeForwardingRule_importBasic (46.46s)
=== RUN   TestAccComputeGlobalAddress_importBasic
--- PASS: TestAccComputeGlobalAddress_importBasic (22.68s)
=== RUN   TestAccComputeHttpHealthCheck_importBasic
--- PASS: TestAccComputeHttpHealthCheck_importBasic (22.94s)
=== RUN   TestAccComputeInstanceTemplate_importBasic
--- PASS: TestAccComputeInstanceTemplate_importBasic (24.15s)
=== RUN   TestAccComputeInstanceTemplate_importIp
--- PASS: TestAccComputeInstanceTemplate_importIp (46.34s)
=== RUN   TestAccComputeInstanceTemplate_importDisks
--- PASS: TestAccComputeInstanceTemplate_importDisks (24.78s)
=== RUN   TestAccComputeInstanceTemplate_importSubnetAuto
--- PASS: TestAccComputeInstanceTemplate_importSubnetAuto (106.00s)
=== RUN   TestAccComputeInstanceTemplate_importSubnetCustom
--- PASS: TestAccComputeInstanceTemplate_importSubnetCustom (97.85s)
=== RUN   TestAccComputeNetwork_importBasic
--- PASS: TestAccComputeNetwork_importBasic (40.15s)
=== RUN   TestAccComputeNetwork_importAuto_subnet
--- PASS: TestAccComputeNetwork_importAuto_subnet (93.64s)
=== RUN   TestAccComputeNetwork_importCustom_subnet
--- PASS: TestAccComputeNetwork_importCustom_subnet (40.68s)
=== RUN   TestAccComputeRoute_importBasic
--- PASS: TestAccComputeRoute_importBasic (66.50s)
=== RUN   TestAccComputeRoute_importDefaultInternetGateway
--- PASS: TestAccComputeRoute_importDefaultInternetGateway (62.41s)
=== RUN   TestAccComputeRouterInterface_import
--- PASS: TestAccComputeRouterInterface_import (222.81s)
=== RUN   TestAccComputeRouterPeer_import
--- PASS: TestAccComputeRouterPeer_import (233.58s)
=== RUN   TestAccComputeRouter_import
--- PASS: TestAccComputeRouter_import (99.28s)
=== RUN   TestAccComputeTargetPool_importBasic
--- PASS: TestAccComputeTargetPool_importBasic (45.92s)
=== RUN   TestAccComputeAddress_basic
--- PASS: TestAccComputeAddress_basic (23.52s)
=== RUN   TestAccComputeAutoscaler_basic
--- PASS: TestAccComputeAutoscaler_basic (70.28s)
=== RUN   TestAccComputeAutoscaler_update
--- PASS: TestAccComputeAutoscaler_update (120.63s)
=== RUN   TestAccComputeBackendBucket_basic
--- PASS: TestAccComputeBackendBucket_basic (28.44s)
=== RUN   TestAccComputeBackendBucket_basicModified
--- PASS: TestAccComputeBackendBucket_basicModified (43.00s)
=== RUN   TestAccComputeBackendBucket_withCdnEnabled
--- PASS: TestAccComputeBackendBucket_withCdnEnabled (28.29s)
=== RUN   TestAccComputeBackendService_basic
--- PASS: TestAccComputeBackendService_basic (68.21s)
=== RUN   TestAccComputeBackendService_withBackend
--- PASS: TestAccComputeBackendService_withBackend (94.16s)
=== RUN   TestAccComputeBackendService_withBackendAndUpdate
--- PASS: TestAccComputeBackendService_withBackendAndUpdate (116.37s)
=== RUN   TestAccComputeBackendService_withConnectionDraining
--- PASS: TestAccComputeBackendService_withConnectionDraining (45.35s)
=== RUN   TestAccComputeBackendService_withConnectionDrainingAndUpdate
--- PASS: TestAccComputeBackendService_withConnectionDrainingAndUpdate (57.82s)
=== RUN   TestAccComputeBackendService_withCDNEnabled
--- PASS: TestAccComputeBackendService_withCDNEnabled (45.14s)
=== RUN   TestAccComputeBackendService_withSessionAffinity
--- PASS: TestAccComputeBackendService_withSessionAffinity (60.54s)
=== RUN   TestAccComputeDisk_basic
--- PASS: TestAccComputeDisk_basic (25.65s)
=== RUN   TestAccComputeDisk_updateSize
--- PASS: TestAccComputeDisk_updateSize (36.65s)
=== RUN   TestAccComputeDisk_fromSnapshotURI
--- PASS: TestAccComputeDisk_fromSnapshotURI (119.82s)
=== RUN   TestAccComputeDisk_encryption
--- PASS: TestAccComputeDisk_encryption (24.32s)
=== RUN   TestAccComputeDisk_deleteDetach
--- PASS: TestAccComputeDisk_deleteDetach (78.62s)
=== RUN   TestAccComputeFirewall_basic
--- PASS: TestAccComputeFirewall_basic (70.39s)
=== RUN   TestAccComputeFirewall_update
--- PASS: TestAccComputeFirewall_update (74.64s)
=== RUN   TestAccComputeForwardingRule_basic
--- PASS: TestAccComputeForwardingRule_basic (45.84s)
=== RUN   TestAccComputeForwardingRule_singlePort
--- PASS: TestAccComputeForwardingRule_singlePort (46.12s)
=== RUN   TestAccComputeForwardingRule_ip
--- PASS: TestAccComputeForwardingRule_ip (46.55s)
=== RUN   TestAccComputeForwardingRule_internalLoadBalancing
--- PASS: TestAccComputeForwardingRule_internalLoadBalancing (67.66s)
=== RUN   TestAccComputeGlobalAddress_basic
--- PASS: TestAccComputeGlobalAddress_basic (22.97s)
=== RUN   TestAccComputeGlobalForwardingRule_basic
--- PASS: TestAccComputeGlobalForwardingRule_basic (111.86s)
=== RUN   TestAccComputeGlobalForwardingRule_update
--- PASS: TestAccComputeGlobalForwardingRule_update (128.64s)
=== RUN   TestAccComputeHealthCheck_tcp
--- PASS: TestAccComputeHealthCheck_tcp (22.68s)
=== RUN   TestAccComputeHealthCheck_tcp_update
--- PASS: TestAccComputeHealthCheck_tcp_update (34.56s)
=== RUN   TestAccComputeHealthCheck_ssl
--- PASS: TestAccComputeHealthCheck_ssl (23.04s)
=== RUN   TestAccComputeHealthCheck_http
--- PASS: TestAccComputeHealthCheck_http (22.81s)
=== RUN   TestAccComputeHealthCheck_https
--- PASS: TestAccComputeHealthCheck_https (22.66s)
=== RUN   TestAccComputeHealthCheck_tcpAndSsl_shouldFail
--- PASS: TestAccComputeHealthCheck_tcpAndSsl_shouldFail (0.01s)
=== RUN   TestAccComputeHttpHealthCheck_basic
--- PASS: TestAccComputeHttpHealthCheck_basic (22.68s)
=== RUN   TestAccComputeHttpHealthCheck_update
--- PASS: TestAccComputeHttpHealthCheck_update (34.86s)
=== RUN   TestAccComputeHttpsHealthCheck_basic
--- PASS: TestAccComputeHttpsHealthCheck_basic (22.91s)
=== RUN   TestAccComputeHttpsHealthCheck_update
--- PASS: TestAccComputeHttpsHealthCheck_update (34.72s)
=== RUN   TestAccComputeImage_basic
--- PASS: TestAccComputeImage_basic (68.30s)
=== RUN   TestAccComputeImage_basedondisk
--- PASS: TestAccComputeImage_basedondisk (80.82s)
=== RUN   TestAccComputeInstanceGroup_basic
--- PASS: TestAccComputeInstanceGroup_basic (83.57s)
=== RUN   TestAccComputeInstanceGroup_update
--- PASS: TestAccComputeInstanceGroup_update (131.04s)
=== RUN   TestAccComputeInstanceGroup_outOfOrderInstances
--- PASS: TestAccComputeInstanceGroup_outOfOrderInstances (84.79s)
=== RUN   TestAccComputeInstanceGroup_network
--- PASS: TestAccComputeInstanceGroup_network (180.65s)
=== RUN   TestAccComputeInstanceTemplate_basic
--- PASS: TestAccComputeInstanceTemplate_basic (24.04s)
=== RUN   TestAccComputeInstanceTemplate_IP
--- PASS: TestAccComputeInstanceTemplate_IP (46.07s)
=== RUN   TestAccComputeInstanceTemplate_networkIP
--- PASS: TestAccComputeInstanceTemplate_networkIP (24.19s)
=== RUN   TestAccComputeInstanceTemplate_disks
--- PASS: TestAccComputeInstanceTemplate_disks (23.83s)
=== RUN   TestAccComputeInstanceTemplate_subnet_auto
--- PASS: TestAccComputeInstanceTemplate_subnet_auto (106.51s)
=== RUN   TestAccComputeInstanceTemplate_subnet_custom
--- PASS: TestAccComputeInstanceTemplate_subnet_custom (104.09s)
=== RUN   TestAccComputeInstanceTemplate_subnet_xpn
--- PASS: TestAccComputeInstanceTemplate_subnet_xpn (89.46s)
=== RUN   TestAccComputeInstanceTemplate_metadata_startup_script
--- PASS: TestAccComputeInstanceTemplate_metadata_startup_script (24.49s)
=== RUN   TestAccComputeInstance_basic_deprecated_network
--- PASS: TestAccComputeInstance_basic_deprecated_network (72.26s)
=== RUN   TestAccComputeInstance_basic1
--- PASS: TestAccComputeInstance_basic1 (62.67s)
=== RUN   TestAccComputeInstance_basic2
--- PASS: TestAccComputeInstance_basic2 (53.39s)
=== RUN   TestAccComputeInstance_basic3
--- PASS: TestAccComputeInstance_basic3 (72.50s)
=== RUN   TestAccComputeInstance_basic4
--- PASS: TestAccComputeInstance_basic4 (56.72s)
=== RUN   TestAccComputeInstance_basic5
--- PASS: TestAccComputeInstance_basic5 (66.70s)
=== RUN   TestAccComputeInstance_IP
--- PASS: TestAccComputeInstance_IP (89.23s)
=== RUN   TestAccComputeInstance_disksWithoutAutodelete
--- PASS: TestAccComputeInstance_disksWithoutAutodelete (83.47s)
=== RUN   TestAccComputeInstance_disksWithAutodelete
--- PASS: TestAccComputeInstance_disksWithAutodelete (64.11s)
=== RUN   TestAccComputeInstance_diskEncryption
--- PASS: TestAccComputeInstance_diskEncryption (78.65s)
=== RUN   TestAccComputeInstance_attachedDisk
--- PASS: TestAccComputeInstance_attachedDisk (74.55s)
=== RUN   TestAccComputeInstance_noDisk
--- PASS: TestAccComputeInstance_noDisk (0.63s)
=== RUN   TestAccComputeInstance_local_ssd
--- PASS: TestAccComputeInstance_local_ssd (66.76s)
=== RUN   TestAccComputeInstance_update_deprecated_network
--- PASS: TestAccComputeInstance_update_deprecated_network (95.33s)
=== RUN   TestAccComputeInstance_forceNewAndChangeMetadata
--- PASS: TestAccComputeInstance_forceNewAndChangeMetadata (145.90s)
=== RUN   TestAccComputeInstance_update
--- PASS: TestAccComputeInstance_update (137.80s)
=== RUN   TestAccComputeInstance_service_account
--- PASS: TestAccComputeInstance_service_account (56.35s)
=== RUN   TestAccComputeInstance_scheduling
--- PASS: TestAccComputeInstance_scheduling (62.29s)
=== RUN   TestAccComputeInstance_subnet_auto
--- PASS: TestAccComputeInstance_subnet_auto (159.37s)
=== RUN   TestAccComputeInstance_subnet_custom
--- PASS: TestAccComputeInstance_subnet_custom (139.88s)
=== RUN   TestAccComputeInstance_subnet_xpn
--- PASS: TestAccComputeInstance_subnet_xpn (145.83s)
=== RUN   TestAccComputeInstance_address_auto
--- PASS: TestAccComputeInstance_address_auto (162.67s)
=== RUN   TestAccComputeInstance_address_custom
--- PASS: TestAccComputeInstance_address_custom (158.96s)
=== RUN   TestAccComputeInstance_private_image_family
--- PASS: TestAccComputeInstance_private_image_family (309.54s)
=== RUN   TestAccComputeInstance_invalid_disk
--- PASS: TestAccComputeInstance_invalid_disk (24.09s)
=== RUN   TestAccComputeInstance_forceChangeMachineTypeManually
--- PASS: TestAccComputeInstance_forceChangeMachineTypeManually (83.52s)
=== RUN   TestAccComputeNetwork_basic
--- PASS: TestAccComputeNetwork_basic (32.21s)
=== RUN   TestAccComputeNetwork_auto_subnet
panic: test timed out after 2h0m0s
```

**Comment by danawillow** [comment]
Yeah, there are a lot of tests and they all take a non-trivial amount of time to run, unfortunately. You might be able to speed it up with a `-parallel n` flag but I've never tried it (and I imagine it might run into quota things. I'm fine with this as-is, since any other failures will get caught by the nightly tests. Merge at will.

**Comment by selmanj** [comment]
Alright. I'll try to take some time to see if there's anything that can be done to speed up some of the tests.

**Pull request merged**